### DISCUSSION TITLE
Force Prisma Client to generate certain Scalar types even if missing in schema?

### DISCUSSION THREAD

**Discussion by haleksandre** [prisma/prisma#21990] [Q&A]
### Question

I was wondering if it is possible to force Prisma to generate some scalar types even if my schema does not use it, for now. The reason is that I want to make a generic GraphQL scalar based on said type.

Here's an example of what I mean

```prisma
// Schema
model Model {
  decimal Decimal?
}
```
would result in Prisma generating types such as `DecimalNullableFilter` & so on since the only `Decimal` field in the schema is nullable. So it wouldn't generate types such as `DecimalFilter` for the same reason as above.

Is there a way to make Prisma generate those types without actually having a field in the schema?

The use case is I am using GraphQL with Pothos & would like to create a GraphQL input type based on `DecimalFilter`. Here's an example of doing this with `StringFilter`:

```ts
import { Prisma } from "@prisma/client";
import { builder } from "./builder.js";

export const StringFilter = builder.inputRef<Prisma.StringFilter>("StringFilter").implement({
	description: "String filter conditions and operators input",
	fields: (t) => ({
		contains: t.field({ type: "String" }),
		equals: t.field({ type: "String" }),
		endsWith: t.field({ type: "String" }),
		startsWith: t.field({ type: "String" }),
		not: t.field({ type: "String" }),
		mode: t.field({ type: QueryMode }),
		search: t.field({ type: "String" }),
		AND: t.field({ type: StringFilter }),
		OR: t.field({ type: StringFilter }),
		NOT: t.field({ type: StringFilter }),
	}),
});
```

**Comment by ludralph** [comment]
Hi @haleksandre 👋 

Thank you for raising this question.

At the moment, there is no method to force Prisma to generate scalar types like `DecimalFilter` without having a corresponding field in the schema. Prisma's type generation is typically based on the actual fields defined in your `schema.prisma` file, as indicated by the way it generates types such as `DecimalNullableFilter` when a nullable `Decimal` field is present.

If you're looking to create a GraphQL input type based on `DecimalFilter`, you might need to define a non-nullable Decimal field in your Prisma schema to have Prisma generate the DecimalFilter type, or you could manually define the input type you need for your GraphQL schema, independent of Prisma's generated types.

However, if you want to see this feature implemented, you can open a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) and document your use case so our Engineers can take a look.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise, please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
